modules/fbs-sql-checker/distance/table_check.py

```diff
@@ -0,0 +1,160 @@
+                        comp_list.append(f.format_comp_db_name(t.value))
+                join_list.append(token.token_first().value)  # append keyword where
+            # extract attributes and iterate through group by clause    
+            if token.ttype == sqlparse.tokens.Keyword and token.value == c.GROUP_BY:
+                j = i + 1
+                while j < len(tokens):
+                    t = tokens[j]
+                    if isinstance(t, sqlparse.sql.Token):
+                        if isinstance(tokens[j], (
+                                sqlparse.sql.IdentifierList, sqlparse.sql.Identifier, sqlparse.sql.Operation,
+                                sqlparse.sql.Function, sqlparse.sql.Parenthesis)):
+                            _extract_group_by_attributes(tokens[j], group_list)
+                            j += 1
+                        # iterating through having tokens
+                        if tokens[j].ttype == sqlparse.tokens.Keyword and tokens[j].value == c.HAVING:
+                            k = j + 1
+                            while k < len(tokens):
+                                if isinstance(tokens[k], sqlparse.sql.Token):
+                                    if isinstance(tokens[k], sqlparse.sql.Comparison):
+                                        having_list.append(f.format_comp_db_name(tokens[k].value))
+                                        k += 1
+                                    if (tokens[k].ttype == sqlparse.tokens.Keyword and tokens[
+                                        k].value == c.ORDER_BY) or (
+                                            tokens[k].ttype == sqlparse.tokens.Punctuation and tokens[k].value == ";"):
+                                        break
+                                k += 1
+                            j += 1
+                        if (t.ttype == sqlparse.tokens.Keyword and (t.value == c.ORDER_BY or t.value == c.HAVING)) or (
+                                t.ttype == sqlparse.tokens.Punctuation and t.value == ";"):
+                            break
+                    j += 1
+            # extract attributes inside order by clause
+            if token.ttype == sqlparse.tokens.Keyword and token.value == c.ORDER_BY:
+                j = i + 1
+                while j < len(tokens):
+                    if isinstance(tokens[j], sqlparse.sql.Token):
+                        if isinstance(tokens[j], sqlparse.sql.IdentifierList):
+                            for t in tokens[j]:
+                                _extract_order_by_attributes(t, order_list)
+                        if isinstance(tokens[j], (
+                                sqlparse.sql.Identifier, sqlparse.sql.Operation, sqlparse.sql.Function,
+                                sqlparse.sql.Parenthesis, sqlparse.sql.Comparison)):
+                            _extract_order_by_attributes(tokens[j], order_list)
+                            j += 1
+                    j += 1
```

Comment:
I would split this function into several functions, because the code is very difficult to read this way. e.g. one function extract{keyword} each (i.e. extractFrom, extractGroupBy, etc.) and possibly split it further. This makes the code difficult to read and very convoluted